CMSgov/design-system
# [WNMGDS-949] - alert component typescript conversion

[Link to ticket](https://jira.cms.gov/browse/WNMGDS-949)

**Please follow the format below and remove any sections that aren't relevant.**
## Summary
Converting the Alert component to typescript. I had to update a few of the Alert tests for typing.

I also updated some other tests that were failing.

## Outstanding Questions
* for the Analytics event object, should any of those properties be optional?

## How to test
* Build this project and confirm that there aren't any build errors
* I tested in mgov child system and locally linked and ran a build -- it was successful
